Gruntfile.js

```diff
@@ -136,11 +138,38 @@ module.exports = (grunt) => {
     }
   }
 
+  async function redeployAndUpdateMainSrc (manifestDir) {
+    if (development) {
+      const { API_URL, GI_PERSIST } = process.env
+      const { dirname, filename } = databaseOptionBags[GI_PERSIST] ?? {}
+      let dest = './data'
+      if (GI_PERSIST === 'sqlite') dest = path.join(dirname, filename)
+      if (GI_PERSIST === 'fs') dest = dirname
+      if (!GI_PERSIST) dest = API_URL
+      grunt.log.writeln(chalk.underline(`Running 'chel deploy to ${dest}'`))
+      const { stdout } = await execWithErrMsg(`./node_modules/.bin/chel deploy ${dest} ${manifestDir}/*.manifest.json`, 'error redeploying contracts')
+      console.log(stdout)
+      const r = /contracts\/([^.]+)\.(?:x|[\d.]+)\.manifest.*\/(.*)/g
+      const manifests = Object.fromEntries(Array.from(stdout.replace(/\\/g, '/').matchAll(r), x => [`gi.contracts/${x[1]}`, x[2]]))
+      await writeFile(manifestJSON, JSON.stringify({ manifests }, null, 2) + '\n')
```

Comment:
This entire section appears to be repeated? If so then can you please place it in a function to DRY it with `deployAndUpdateMainSrc`? Or, alternatively (perhaps even better), merge the two functions into 1 function and pass in a parameter to distinguish behavior.

Reply:
Yea, that's the code duplication I was talking about on Slack